src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final double arrayVolumeAdd = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object [DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
 
+        if (size == elementData.length) {
+            grow(size);
+        }
+        elementData[size] = value;
+        size++;
+
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, 0, size);
+        if (size == elementData.length) {
+            grow(size);
+        }
+        System.arraycopy(elementData,index,elementData,index + 1,size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
 
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, 0, size - 1);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, 0, size - 1);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, 0, size - 1);
+        final T element = (T)elementData[index];
+        int numMoved = size - index - 1;
+        if (numMoved > 0) {
+            System.arraycopy(elementData, index + 1, elementData, index,
+                    numMoved);
+        }
```

Comment:
arraycopy won't work anyway if numMoved is 0
```suggestion
         System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
        }
```

Reply:
not fixed